enterprise/server/raft/cache/cache.go

```diff
@@ -86,6 +97,26 @@ func (rc *RaftCache) Create(nhid string, streamConnections uint64, v dbConfig.Ta
 	return r, nil
 }
 
+func Register(env *real_environment.RealEnv) error {
+	if *listenAddr == "" {
+		return nil
+	}
+	rcConfig := &Config{
+		RootDir:       *rootDirectory,
+		ListenAddress: *listenAddr,
+		Join:          *join,
+		HTTPPort:      *httpPort,
+		GRPCPort:      *gRPCPort,
+	}
+	rc, err := NewRaftCache(env, rcConfig)
+	if err != nil {
+		return status.InternalErrorf("Error enabling raft cache: %s", err.Error())
+	}
+	defer rc.Stop()
```

Comment:
defers are executed right before the function returns, so this will actually stop the cache before Register returns. I don't think that's what we want. The reason this was OK in main before is because main loops forever until exit.

You could instead register a graceful shutdown function (using the healthchecker in the env) that calls rc.Stop?

Reply:
Thanks for the catch, fixed!